controllers/hyperfoil_controller.go

```diff
@@ -543,46 +543,15 @@ func compareControllerPod(i1 interface{}, i2 interface{}, logger logr.Logger) bo
 		logger.Info("Cannot cast to Pods: " + fmt.Sprintf("%v | %v", i1, i2))
 		return false
 	}
-	c1 := p1.Spec.Containers[0]
-	c2 := p2.Spec.Containers[0]
-	if c1.Image != c2.Image {
-		logger.Info("Images don't match: " + c1.Image + " | " + c2.Image)
-		return false
-	}
-	if !reflect.DeepEqual(c1.Command, c2.Command) {
-		logger.Info("Commands don't match: " + fmt.Sprintf("%v | %v", c1.Command, c2.Command))
-		return false
-	}
-	if len(c1.Env) != len(c2.Env) {
-		logger.Info("Env vars differ " + fmt.Sprintf("%v vs %v", len(c1.Env), len(c2.Env)))
-		return false
-	}
-	for i, ev1 := range c1.Env {
-		ev2 := c2.Env[i]
-		if !reflect.DeepEqual(ev1, ev2) {
-			return false
-		}
-	}
 
-	if len(c1.EnvFrom) != len(c2.EnvFrom) {
-		logger.Info("Env vars differ " + fmt.Sprintf("%v vs %v", len(c1.EnvFrom), len(c2.EnvFrom)))
-		return false
-	}
-	for i, ef1 := range c1.EnvFrom {
-		ef2 := c2.EnvFrom[i]
-		if ef1.SecretRef == nil || ef2.SecretRef == nil || ef1.SecretRef.LocalObjectReference.Name != ef2.SecretRef.LocalObjectReference.Name {
-			logger.Info("Secrets for env vars don't match.")
-			return false
-		}
-	}
-	if !compareVolume(p1.Spec.Volumes, p2.Spec.Volumes, "log", logger) ||
-		!compareVolume(p1.Spec.Volumes, p2.Spec.Volumes, "prehooks", logger) ||
-		!compareVolume(p1.Spec.Volumes, p2.Spec.Volumes, "posthooks", logger) ||
-		!compareVolume(p1.Spec.Volumes, p2.Spec.Volumes, "hyperfoil", logger) {
-		return false
+	// Compare for equality only whatever field is set in p1.Spec.
+	if equality.Semantic.DeepDerivative(p1.Spec, p2.Spec) {
+		return true
 	}
 
-	return true
+	diff := cmp.Diff(p1.Spec, p2.Spec)
+	logger.Info("Pod " + p1.GetName() + " diff (-want,+got):\n" + diff)
+	return false
```

Comment:
This isn't strictly required by it's a nice way of logging what are the differences.
I can remove them if it's not wanted here.